datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1237,12 +1231,7 @@ fn ensure_distribution(
                     child = add_spm_on_top(child);
```

Comment:
@Dandandan Do we need the additional round robin for single partition? This looks unnecessary as single partitioning is not costly as hash partitioning.

Reply:
Let me focus on `repartition_file_scans` change first in this PR. I will open another PR to clean up these round robin stuffs.